internal/consensus/reactor.go

```diff
@@ -1192,34 +1209,31 @@ func (ps *PeerState) setHasProposalBlockPart(height int64, round int32, index in
 	ps.PRS.ProposalBlockParts.SetIndex(index, true)
 }
 
-// PickSendVote picks a vote and sends it to the peer.
-// Returns true if vote was sent.
-func (ps *PeerState) PickSendVote(votes types.VoteSetReader) bool {
-	if vote, ok := ps.PickVoteToSend(votes); ok {
-		ps.logger.Debug("Sending vote message", "ps", ps, "vote", vote)
-		if ps.peer.Send(p2p.Envelope{
-			ChannelID: VoteChannel,
-			Message: &cmtcons.Vote{
-				Vote: vote.ToProto(),
-			},
-		}) {
-			ps.SetHasVote(vote)
-			return true
-		}
-		return false
+// SendVoteSetHasVote sends the vote to the peer.
+// Returns true and marks the peer as having the vote if the vote was sent.
+func (ps *PeerState) SendVoteSetHasVote(vote *types.Vote) bool {
+	ps.logger.Debug("Sending vote message", "ps", ps, "vote", vote)
+	if ps.peer.Send(p2p.Envelope{
```

Comment:
Can we simplify this if call?

Reply:
We can, but I propose to do this in another refactoring effort, as these calls are done like this all over our code